apps/core/src/hooks/useGetClockTimestamp.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { useIotaClient } from '@iota/dapp-kit';
+import { IotaClient } from '@iota/iota-sdk/client';
+import { useQuery } from '@tanstack/react-query';
+
+const CLOCK_PACKAGE_ID = '0x06';
+
+type ClockFields = {
+    id: {
+        id: string;
+    };
+    timestamp_ms: string;
+};
+
+export function useGetClockTimestamp() {
+    const client = useIotaClient();
+    return useQuery({
+        queryKey: ['get-clock-timestamp', client],
+        queryFn: async () => {
+            return getClockTimestamp(client);
+        },
+        staleTime: 10 * 1000,
+        refetchInterval: 10 * 1000, // refetch every 10 seconds to keep the clock updated but not overload the server
+    });
```

Comment:
If you really don't want to have TS issues I guess you could do something like this right?
```suggestion
		initialData: () => 0 // Or Infinity ?
    });
```

Reply:
For such case I'd use placeholder data as it's not caching and we don't need to wait 10s until get real data

Reply:
Ohh, very fair point